redi-connect-backend/scripts/20210301_mentoring-topics-consolidation-email-to-mentors.js

```diff
@@ -0,0 +1,152 @@
+ 
+<mj-text mj-class="text" padding="0 0 10px 0">So, what changed?</mj-text>
+<mj-text mj-class="text" padding="0 0 10px 0">👉 We added a few more categories to better group the different mentoring topics, i.e. you will now see the categories: Software Engineering, Career Support, 🆕 Design, 🆕 Other Professions, 🆕 Language and 🆕 Other.</mj-text>
+<mj-text mj-class="text" padding="0 0 10px 0">👉 We created more topics so that you can be more specific, esp. under the category Career Support, but also if you work in a non-tech position.</mj-text>
+<mj-text mj-class="text" padding="0 0 10px 0">👉 A few old topics were dropped since not many people selected them. Other topics we split up into separate ones, e.g. Mobile Development, Web Development and Graphics & UX/UI (Berlin users only). Whenever that was the case, we selected all new options for you. So please have a look and make sure which topics you would like to keep checked.</mj-text>
+<mj-text mj-class="text" padding="0 0 20px 0">👉 Since overall, we added more topics, mentees will now be able to select up to four (previously three) mentoring topics they're looking for support in.</mj-text>
+ 
+<mj-text mj-class="text" padding="0 0 20px 0">We hope that these changes will allow for better matches to be formed on ReDI Connect. And please let us know if you feel we missed any major mentoring subject.</mj-text>
+
+<mj-text mj-class="text" padding="0 0 20px 0"><a href="https://connect.redi-school.org">ReDI? Log into your profile and check out the new topics.<a></mj-text>
+ 
+<mj-text mj-class="text" padding="0 0 20px 0">Happy mentoring!</mj-text>
+ 
+<mj-text mj-class="text" padding="0 0 5px 0">Best,</mj-text>
+<mj-text mj-class="text" padding="0 0 0 0">Miriam</mj-text>
+`
+);
+
+const emailBodyParsed = mjml2html(emailBodyMjml, {
+  filePath: path.resolve(__dirname, '..', 'lib', 'email', 'templates'),
+});
+
+const sendMjmlEmail = Rx.bindNodeCallback(transporter.sendMail.bind(transporter));
+
+/* eslint-disable-next-line */
+const sendMentorEmail = ({ recipient, firstName }) => {
+  const mailOptions = {
+    from: 'career@redi-school.org',
+    replyTo: 'miriam@redi-school.org',
+    to: recipient,
+    subject: 'ReDI, Set, Go! Changes to mentoring topics in ReDI Connect and what you should do',
+    html: emailBodyParsed.html.replace(':firstName:', firstName),
+  };
+
+  return sendMjmlEmail(mailOptions);
+};
+
+const redUserFind = (q) => bindNodeCallback(RedUser.find.bind(RedUser))(q);
+
+redUserFind({ include: 'redProfile' })
+  .pipe(
+    delay(1000),
+    switchMap((users) => from(users)),
+    map((data) => {
+      const pojo = JSON.parse(JSON.stringify(data));
```

Comment:
If you had a moment could you explain what makes us need both `parse` and `stringify` in this case?

Reply:
It's an old `loopback` gotcha that this works around quite simply. I've re-used the pattern in a few dozen email-sending scripts. 

Thankfully our use of Loopback is reaching the end of its lifetime now that Salesforce and (likely) NestJS are in sight, so we'll eventually be come up with a new template for email-sending scripts.